numba_cuda/numba/cuda/dispatcher.py

```diff
@@ -2119,6 +2125,18 @@ def _reduce_states(self):
         return dict(py_func=self.py_func, targetoptions=self.targetoptions)
 
 
+def register_arg_handler(handler, handled_types, impl):
+    global _arg_handlers
+
+    for ty in handled_types:
+        typeof_impl.register(ty)(impl)
+        if _arg_handlers.get(ty, None):
```

Comment:
This should come before registering the `typeof_impl`, because the registration can't be backed out afterwards if something goes wrong here.